### ISSUE TITLE
postgres db backend connection pooling

### ISSUE THREAD

**Issue by poblin-orange** [rancher/kine#63]
Hello,
using k3s (single server + 2 agents) with an external postgresdb
Im encountering an (unrelated) issue with very frequent dns lookup to our dns servers.

Zooming in, i see kine is opening /closing many connections. Is there a way to configure datastore to pool postgresql connections ?

**Comment by brandond** [comment]
We have added the capability to tune the connection pooling parameters to Kine, but have not yet exposed these options through k3s.

**Comment by poblin-orange** [comment]
Thks for your answer. Seems the pgx library has  this ability. https://github.com/JackC/pgx
Any way we could pass this as datastore url and avoid complexifying the k3s command-line ?

**Comment by brandond** [comment]
No, we're not using that library. We're using github.com/lib/pq.

**Comment by poblin-orange** [comment]
Ok. note sure how related it is, but switching to mysql gave us a huge perf improvment.

**Comment by brandond** [comment]
I took a shot at improving compaction performance with the most recent round of updates but I don't think we've baseline benchmarked database performance. Can I ask for more info on the database platforms you switched from and to?

**Comment by poblin-orange** [comment]
on premise vms databases, provisionned with bosh release:
- from https://github.com/cloudfoundry/postgres-release/releases/tag/v42 (postgres 11.9)
- to https://github.com/cloudfoundry-incubator/pxc-release/releases/tag/v0.31.0 (mysql v5.7.31)

Single vms db, 2vcpu, 8Go memory.

**Comment by brandond** [comment]
Are you just looking at database connections and connection rate to gauge performance? Are you evaluating any of the apiserver datastore performance metrics?

**Comment by poblin-orange** [comment]
no détail monitoring, just saw cpu load improvement
